pkg/controller/storagecluster/noobaa_system_reconciler.go

```diff
@@ -42,63 +32,38 @@ func (r *ReconcileStorageCluster) ensureNoobaaSystem(sc *ocsv1.StorageCluster, r
 		}
 		return err
 	}
-
-	if foundCeph.Status.State == cephv1.ClusterStateCreated {
-		cephClusterCreated = true
+	if foundCeph.Status.State != cephv1.ClusterStateCreated {
+		reqLogger.Info("Waiting on ceph cluster to initialize before starting noobaa")
+		return nil
 	}
 
-	// check if this noobaa instance aleady exists
-	found := &nbv1.NooBaa{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: nb.ObjectMeta.Name, Namespace: sc.Namespace}, found)
+	// Take ownership over the noobaa object
+	nb := &nbv1.NooBaa{}
+	err := controllerutil.SetControllerReference(sc, nb, r.scheme)
```

Comment:
`err` should be defined above.

```suggestion
	err = controllerutil.SetControllerReference(sc, nb, r.scheme)
```

Reply:
fixed